### DISCUSSION TITLE
Set x-xsrf-token and send it in header during request RTK Query

### DISCUSSION THREAD

**Discussion by Neo363** [reduxjs/redux-toolkit#3034] [Community Recipes]
I have Laravel API + NextJs application with Sanctum authentication.

With Sanctum before sending request to the route, you need to send a request to /sanctum/csrf-cookie 'to initialize csrf protection'.

Here i how my RTK Query API looks:
![vcq4T](https://user-images.githubusercontent.com/57476799/209515036-4cb5dd66-69b2-4fd1-9a44-6423bcd9c0cb.png)

Here is how the functions are called:
![MWFwJ](https://user-images.githubusercontent.com/57476799/209515067-36037651-66af-4fcc-82dc-1aaa29e83b2e.png)

The logic works fine but i wanted to use only RTK Query and not depend on axios. Also, is there any way to handle getting the csrf-cookie and send it during the request from `createAPI()`?

**Comment by sobyf** [comment]
> I have Laravel API + NextJs application with Sanctum authentication.
> 
> With Sanctum before sending request to the route, you need to send a request to /sanctum/csrf-cookie 'to initialize csrf protection'.
> 
> Here i how my RTK Query API looks: ![vcq4T](https://user-images.githubusercontent.com/57476799/209515036-4cb5dd66-69b2-4fd1-9a44-6423bcd9c0cb.png)
> 
> Here is how the functions are called: ![MWFwJ](https://user-images.githubusercontent.com/57476799/209515067-36037651-66af-4fcc-82dc-1aaa29e83b2e.png)
> 
> The logic works fine but i wanted to use only RTK Query and not depend on axios. Also, is there any way to handle getting the csrf-cookie and send it during the request from `createAPI()`?

do you find any solution?i have this problem too

**Reply by Neo363** [reply]
I haven't found any solutions yet.

**Comment by nicolasborromeo** [comment]
I have a similar issue with the csrf token not being recognized by the database when I send the request
